src/common/transformations/src/transformations/utils/utils.cpp

```diff
@@ -499,7 +499,7 @@ bool is_constant_and_all_values_equal_int(const Output<Node>& output, const int6
     return false;
 }
 
-bool is_on_constant_path(const ov::Output<ov::Node>& output) {
+bool is_on_constant_path(const ov::Output<ov::Node>& output, bool params_allowed) {
```

Comment:
This function has a specific naming (is_on_constant_path) and it is better to refactor it as template:
template <typename... AllowedTypes>
bool is_on_path(const ov::Output<ov::Node>& output) {...}